app/queries/patient_summary_query.rb

```diff
@@ -18,17 +18,27 @@ def self.label_for(filter)
     FILTERS.fetch(filter)
   end
 
-  def initialize(relation: PatientSummary.all, filters: [])
-    @relation = relation
+  def initialize(only_overdue: true, assigned_facility: nil, next_appointment_facilities: Facility.none, filters: [])
+    @only_overdue = only_overdue
+    @relation = PatientSummary
+    @relation = @relation.where(next_appointment_facility_id: next_appointment_facilities) if next_appointment_facilities.present?
+    @relation = @relation.where(assigned_facility_id: assigned_facility.id) if assigned_facility
     @filters = filters
   end
 
   def call
-    result = if filters.include?("only_less_than_year_overdue")
-      relation.overdue
+    result = if @only_overdue
+      if filters.include?("only_less_than_year_overdue")
+        relation.overdue
+      else
+        relation.all_overdue
+      end
+    elsif filters.include?("only_less_than_year_overdue")
+      relation.last_year_unvisited
```

Comment:
The filtering should be applied/translated to scopes in the controller. Passing the filters as strings leaks in controller concerns. This is a larger change, can tackle in a follow up PR.